elasticdl/python/worker/data_shard_service.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2020 The ElasticDL Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import threading
+from collections import deque
+
+from elasticdl.proto import elasticdl_pb2
+from elasticdl.python.common.constants import TaskExecCounterKey
+from elasticdl.python.common.grpc_utils import build_channel
+from elasticdl.python.worker.master_client import MasterClient
+
+
+class DataShardService(object):
+    def __init__(
+        self, batch_size, master_client=None,
+    ):
+        if master_client is None:
+            master_addr = os.getenv("MASTER_ADDR")
+            worker_id = int(os.getenv("WORKER_ID"))
+            self._mc = MasterClient(build_channel(master_addr), worker_id)
+        else:
+            self._mc = master_client
+
+        self._batch_size = batch_size
+        self._lock = threading.Lock()
+        self._failed_record_count = 0
+        self._reported_record_count = 0
+        self._current_task = None
+        self._pending_tasks = deque()
+
+    def get_current_task(self):
+        return self._current_task
+
+    def get_task(self, task_type=None):
+        task = self._mc.get_task(task_type)
+        if task.shard.name and task.type == elasticdl_pb2.TRAINING:
```

Comment:
We may not depend on `task.shard.name` to determine if there are new tasks or not.

Reply:
Yes, I add a todo to use task type STOP to determine whether there are new tasks or not. And I will commit the modification in the next PR.